From 7be73bbc5ca3424be1c5462ec9f86f6b8ca5b67a Mon Sep 17 00:00:00 2001
From: John Thiltges <jthiltges2@unl.edu>
Date: Thu, 11 Dec 2025 10:58:57 -0600
Subject: [PATCH 1/2] Remove the checks requiring a bind_dn

The code assumes an Active Directory server. For a non-AD server such as OpenLDAP, it works fine with an anonymous bind.
---
 pkgs/duoauthproxy/duoauthproxy/lib/util.py                | 5 -----
 .../duoauthproxy/lib/validation/config/check/ad_client.py | 8 ++++----
 2 files changed, 4 insertions(+), 9 deletions(-)

diff --git a/pkgs/duoauthproxy/duoauthproxy/lib/util.py b/pkgs/duoauthproxy/duoauthproxy/lib/util.py
index 19bfc73..487fe32 100644
--- a/pkgs/duoauthproxy/duoauthproxy/lib/util.py
+++ b/pkgs/duoauthproxy/duoauthproxy/lib/util.py
@@ -532,11 +532,6 @@ def parse_ad_client(config):
     else:
         ldap_filter = None
 
-    # A blank bind_dn will be rejected with auth-type plain in validation
-    # otherwise we supply a default
-    if not bind_dn:
-        bind_dn = "<ROOT>"
-
     warning_message = warn_insecure_settings(auth_type, transport_type)
     if warning_message:
         log.msg(warning_message)
diff --git a/pkgs/duoauthproxy/duoauthproxy/lib/validation/config/check/ad_client.py b/pkgs/duoauthproxy/duoauthproxy/lib/validation/config/check/ad_client.py
index 16fedab..a77eb56 100644
--- a/pkgs/duoauthproxy/duoauthproxy/lib/validation/config/check/ad_client.py
+++ b/pkgs/duoauthproxy/duoauthproxy/lib/validation/config/check/ad_client.py
@@ -96,8 +96,8 @@ def check_config_values(config, toolbox):
     problems = []
     config_test_resolver = base.get_basic_config_resolver(toolbox)
     config_dict = {
-        "service_account_username": toolbox.test_is_string,
-        "service_account_password": toolbox.test_is_string,
+        "service_account_username": functools.partial(toolbox.test_is_string, allow_empty=True),
+        "service_account_password": functools.partial(toolbox.test_is_string, allow_empty=True),
         "service_account_password_protected": toolbox.test_is_string,
         "search_dn": toolbox.test_dn,
         "security_group_dn": toolbox.test_dn,
@@ -132,7 +132,7 @@ def check_config_values(config, toolbox):
         config, config_test_resolver, dynamic_test_resolver
     )
 
-    problems += check_valid_service_account_username_for_auth_type(config, toolbox)
+    #problems += check_valid_service_account_username_for_auth_type(config, toolbox)
 
     problems += base.run_config_value_checks(config, config_test_resolver)
 
@@ -156,7 +156,7 @@ def check_config_dependencies(config, toolbox):
 
     """
     problems = check_valid_cert_for_transport(config, toolbox)
-    problems += check_valid_bind_dn_for_auth_type(config, toolbox)
+    #problems += check_valid_bind_dn_for_auth_type(config, toolbox)
     problems += check_hostname_verification(config, toolbox)
     return problems
 
-- 
2.52.0

